lib/types.ts

```diff
@@ -50,6 +50,12 @@ export type VerificationState =
   | 'unverified'
   | 'verified';
 
+export type Collection =
+  | { type: 'all' }
+  | { type: 'trash' }
+  | { type: 'tag'; tagHash: TagHash | null }
```

Comment:
previously I might have hinted that this should be the tag hash, so if I did I apologize. based on how we use this I think we'll want to store the `tagName` instead, since we display it.

if we _do_ want to store the `tagHash` to make sure we're consistent all over, then we need to perform a lookup to display the tag name, something we could use a selector for, if it doesn't already exist.

`No notes tagged "${tagName(collection.tagHash)}"`

Reply:
To be honest you didn't; I just set it to `TagHash` because `openedTag` was a `TagHash`. I think selectors make sense, with it as a `Tag` instead.

Reply:
it's easily to collapse a tag name into a tag hash than to go backwards, since collapsing to the hash can be done with just the name but going from the hash to the name involves a lookup in state.

Reply:
> I think selectors make sense, with it as a Tag instead.

Note that `Tag` is a type we don't use too often inside the UI. `TagName` is what we use for its namesake. Fun fact! it's a "branded type" meaning that the type system won't let you assign arbitrary strings (or a tag hash) to a value that's typed as a `TagName` unless you declare it `as TagName`. This was introduced to remove the confusion and opportunities for bugs around treating the two as if they are the same.

Reply:
Set as `TagName` instead. :)